src/services/transaction.rs

```diff
@@ -41,16 +51,46 @@ impl From<transaction::TransactionModelError> for TransactionServiceError {
     }
 }
 
+impl From<user::UserModelError> for TransactionServiceError {
+    fn from(error: user::UserModelError) -> Self {
+        TransactionServiceError::UserModelFailed(error)
+    }
+}
```

Comment:
You could try some crates such as [snafu](https://github.com/shepmaster/snafu), [thiserror](https://crates.io/crates/thiserror) or [anyhow](https://docs.rs/anyhow/latest/anyhow/).

Reply:
Good suggestion but dealing with errors manually is not a big deal for now so I'm not going to use it.